Tests/GeoLib/TestLineSegmentIntersect2d.cpp

```diff
@@ -0,0 +1,129 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/LICENSE.txt
+ */
+
+#include "gtest/gtest.h"
+#include <array>
+#include <ctime>
+#include <functional>
+#include <memory>
+#include <random>
+
+#include "Tests/GeoLib/AutoCheckGenerators.h"
+
+#include "MathLib/Point3d.h"
+#include "GeoLib/AnalyticalGeometry.h"
+
+namespace ac = autocheck;
+
+class LineSegmentIntersect2dTest : public testing::Test
+{
+public:
+	using PointGenerator = ac::RandomCirclePointGeneratorXY<ac::generator<double>>;
+	using SymmSegmentGenerator = ac::SymmSegmentGeneratorXY<PointGenerator>;
+	using PairSegmentGenerator = ac::PairSegmentGeneratorXY<SymmSegmentGenerator>;
+
+	PointGenerator point_generator1 = PointGenerator(
+	    MathLib::Point3d(std::array<double, 3>{{0.0, 0.0, 0.0}}), 1.0);
+	SymmSegmentGenerator segment_generator1 = SymmSegmentGenerator{point_generator1,
+		std::bind(ac::reflect, point_generator1.center, std::placeholders::_1)};
+
+	PointGenerator point_generator2 = PointGenerator(
+	    MathLib::Point3d(std::array<double, 3>{{2.0, 0.0, 0.0}}), 1.0);
+	SymmSegmentGenerator segment_generator2 = SymmSegmentGenerator{point_generator2,
+		std::bind(ac::reflect, point_generator2.center, std::placeholders::_1)};
+
+	MathLib::Vector3 const translation_vector1 = {2,2,0};
+	PairSegmentGenerator pair_segment_generator1 = PairSegmentGenerator{
+	    segment_generator1,
+	    std::bind(ac::translate, translation_vector1, std::placeholders::_1)};
+
+	MathLib::Vector3 const translation_vector2 = {0,0,0};
+	PairSegmentGenerator pair_segment_generator2 = PairSegmentGenerator{
+	    segment_generator1,
+	    std::bind(ac::translate, translation_vector2, std::placeholders::_1)};
+
+	ac::gtest_reporter gtest_reporter;
+};
+
+TEST_F(LineSegmentIntersect2dTest, RandomSegmentOrientationIntersecting)
```

Comment:
Please add a short description what setup is tested and what we expect. Not a math proof but one sentence.